internal/testutils/jimmtest/env.go

```diff
@@ -127,10 +127,27 @@ func (e *Environment) User(name string) *User {
 func (u User) addUserRelations(c *qt.C, jimmTag names.ControllerTag, db *db.Database, client *openfga.OFGAClient) {
 	if u.ControllerAccess == "superuser" {
 		dbUser := u.DBObject(c, db)
-		u := openfga.NewUser(&dbUser, client)
-		err := u.SetControllerAccess(context.Background(), jimmTag, ofganames.AdministratorRelation)
+		openfgaUser := openfga.NewUser(&dbUser, client)
+		err := openfgaUser.SetControllerAccess(context.Background(), jimmTag, ofganames.AdministratorRelation)
 		c.Assert(err, qt.IsNil)
 	}
+	if len(u.CanAddModel) > 0 {
+		dbUser := u.DBObject(c, db)
+		openfgaUser := openfga.NewUser(&dbUser, client)
+
+		for _, controllerUUID := range u.CanAddModel {
+			if names.IsValidController(controllerUUID) == false {
+				c.Fatalf("invalid controller UUID for can-add-model: %s", u.CanAddModel)
+			}
+
+			err := openfgaUser.SetControllerAccess(
+				context.Background(),
+				names.NewControllerTag(controllerUUID),
+				ofganames.CanAddModelRelation,
+			)
+			c.Assert(err, qt.IsNil)
+		}
+	}
```

Comment:
You can delete this as the permissions for the resource are normally defined on the resource itself rather than on the user.

Reply:
See in this PR (not merged yet) https://github.com/canonical/jimm/pull/1745/files#diff-4cc4a751474a3a0cd8bbdd149ded07c916c37c380880bfdc2487f6a827688cb7
Then you can do the following in the yaml for the env,
```
  users:
  - user: alice@canonical.com
    access: add-model
```
This just matches the pattern elsewhere.